ang/civicase/activity/directives/add-activity-menu.directive.js

```diff
@@ -21,27 +21,14 @@
 
     (function init () {
       if (_.isEmpty($scope.case.activity_count)) {
-        $scope.case.activity_count = getActivitiesCount();
+        $scope.case.activity_count = {};
```

Comment:
@deb1990 I'm trying to wrap my head around this, but can't seem to understand it. At first we were using JS to count the number of activity types, then we removed this so they are counted in the BE. Where is the count stored when it comes from the BE?

Why are we passing an empty object to `getAvailableActivityTypes` ? Wouldn't this break the feature that we should not add more than X amount of a certain activity type?

Reply:
@reneolivo So, previously, `getActivitiesCount()` always resulted in an empty object, because `allActivities` was empty.
So we are not introducing any regression, because the end result is the same.
Also when `$scope.case.activity_count` is empty, `$scope.availableActivityTypes` is expected to have all the activity types. And I verified, and it is still have functionality.

Reply:
Then we should remove `$scope.case.activity_count` if we don't use it for the count. We

Also, what about this block? 
https://github.com/compucorp/uk.co.compucorp.civicase/blob/0a31a1adfdc9bb9255bddf410397f59d05979160/ang/civicase/activity/directives/add-activity-menu.directive.js#L50

Shouldn't re refactor it so it uses `$scope.availableActivityTypes` instead?